src/test/regress/expected/multi_prune_shard_list.out

```diff
@@ -203,3 +203,27 @@ SELECT print_sorted_shard_intervals('pruning_range');
  {800004,800005,800006,800007}
 (1 row)
 
+-- ===================================================================
+-- test pruning using values whose types are coerced
+-- ===================================================================
+CREATE TABLE coerce_hash (
+	id bigint NOT NULL,
+	value text NOT NULL
+);
+SELECT create_distributed_table('coerce_hash', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- PostgreSQL will cast id to numeric rather than 1.0 to bigint...
+-- We used to blindly pass the RHS' Datum to our comparison func.,
+-- resulting in inaccurate pruning. An Assert now verifies type-
+-- compatibility; the following would crash the server in an Assert-
+-- before the underlying issue was addressed. It looks like a boring
+-- test now, but if the old behavior is restored, it should crash again.
+SELECT * FROM coerce_hash WHERE id = 1.0::numeric;
```

Comment:
I just hesitated to add a "do a bunch of `SELECTs`" test because (a) it's not guaranteed to fail in the case of a regression, and (b) I don't really like adding tests that do things multiple times (our tests take long enough as it is). But I figure it won't actually take that long. I'll add it.